frontend/app/modules/chat/chat-service.js

```diff
@@ -0,0 +1,44 @@
+(function() {
+  angular
+  .module('chat-app.chat')
+  .service('ChatService', ['$http', Service]);
+
+  function Service($http) {
+    var service = this;
+
+    service.getChatMessages = function(channel) {
+      return $http(
+        {
+          method: "GET",
+          url: "/api/getChannelMessages?channelId=" + channel.channelId,
+          headers: {
+            Authorization: "Bearer " + Cookies.get("auth"),
+          }
+        })
+        .then(function(res) {
+          const messagesArray = [];
+
+          res.data.data.forEach(function(val) {
+            val.contents = JSON.parse(val.contents); //Parses the stringified string
+            messagesArray.push(val);
```

Comment:
I have sent the chat message as a JSON.stringified string. So e.g. '"hello \n raf"'. This way I could preserve formatting like new lines.